test/integration/node-specific/abort_signal.test.ts

```diff
@@ -0,0 +1,872 @@
+                await clientEncryption.createEncryptedCollection(
+                  utilClient.db('abortSignal'),
+                  'support',
+                  {
+                    provider: 'local',
+                    createCollectionOptions: {
+                      encryptedFields: autoEncryption.encryptedFieldsMap['abortSignal.support']
+                    }
+                  }
+                )
+              ).encryptedFields;
+            } finally {
+              await utilClient.close();
+            }
+
+            utilClient = this.configuration.newClient({}, { autoEncryption });
+            try {
+              await utilClient
+                .db('abortSignal')
+                .collection('support')
+                .insertMany([
+                  { a: 1, ssn: '0000-00-0001' },
+                  { a: 2, ssn: '0000-00-0002' },
+                  { a: 3, ssn: '0000-00-0003' }
+                ]);
+            } finally {
+              await utilClient.close();
+            }
+          });
+
+          beforeEach(async function () {
+            client = this.configuration.newClient(
+              {},
+              {
+                autoEncryption,
+                monitorCommands: true,
+                appName: 'abortSignalClient',
+                __enableMongoLogger: true,
+                __internalLoggerConfig: { MONGODB_LOG_SERVER_SELECTION: 'debug' },
+                mongodbLogPath: { write: log => logs.push(log) },
+                serverSelectionTimeoutMS: 10_000,
+                maxPoolSize: 1
+              }
+            );
+            await client.connect();
+            db = client.db('abortSignal');
+            collection = db.collection('support');
+
+            method = collection[cursorName].bind(collection);
+            filter = cursorName === 'aggregate' ? [] : {};
+          });
+
+          afterEach(async function () {
+            await client?.close();
+          });
+
+          describe('and the signal is aborted during command encryption', fleMetadata, () => {
+            function test(cursorAPI, args) {
+              let controller: AbortController;
+              let signal: AbortSignal;
+              let cursor: AbstractCursor<{ a: number }>;
+
+              beforeEach(async function () {
+                controller = new AbortController();
+                signal = controller.signal;
+                cursor = method(filter, { signal });
+              });
+
+              afterEach(async function () {
+                sinon.restore();
+                await cursor?.close();
+              });
+
+              it(`rejects ${cursorAPI.toString()}`, fleMetadata, async () => {
+                const willBeResultBlocked = captureCursorAPIResult(cursor, cursorAPI, args);
+
+                const stub = sinon
+                  .stub(client.options.autoEncrypter, 'encrypt')
+                  .callsFake(function (...args) {
+                    controller.abort();
```

Comment:
I don't think this test is actually testing that the signal is aborted during `encrypt()` - instead it tests that if the signal has been aborted when `encrypt()` is called, `encrypt()` rejects with a DOMException. 

Do we think this is necessary to test?  Seems like we have coverage for this with the state machine tests.

Reply:
I would happy to remove these. So much to set up, so little value 😭 I agree the state machine tests have the coverage we need.